### ISSUE TITLE
What is "relative-link" provided by?

### ISSUE THREAD

**Issue by parkerlreed** [SmartisanTech/Wrench#1]
Line 30 of build-linux.sh calls `relative-link` but I can't find any information about this command.

Arch x86_64

**Comment by baohaojun** [comment]
Sorry, relative-link is a script from my system-config project, https://github.com/baohaojun/system-config

**Comment by parkerlreed** [comment]
Any reason that the system `adb` cannot be used?

````
[parker@x3720 Wrench]$ ./build-linux.sh 
rsync: link_stat "/home/parker/build/Wrench/adb" failed: No such file or directory (2)
rsync: link_stat "/home/parker/build/Wrench/the-true-adb" failed: No such file or directory (2)
symlink has no referent: "/home/parker/build/Wrench/linux/binaries/the-true-adb"
symlink has no referent: "/home/parker/build/Wrench/linux/binaries/adb"
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1178) [sender=3.1.2]
[parker@x3720 Wrench]$ ls
adb                       bhj_help.hpp        contactmodel.h      fileopenfilter.cpp     linux                 painterpathitem.h      qcellphonetextedit.cpp  resources             uc-to-py.tbl            Wrench.ico
adb-client                build-all.sh        contacts.lua        fileopenfilter.h       lua                   painterrectitem.cpp    qcellphonetextedit.h    run-on-vbox-xp.sh     ui_promptdialog.h       wrench.lua
adbclient.cpp             build-linux.sh      dialoggetentry.cpp  filteringedit.cpp      luaexecutethread.cpp  painterrectitem.h      qianghongbao.twa        screenshot_shell.png  update-md5s.sh          wrenchmainwindow.cpp
adbclient.h               build-mac.sh        dialoggetentry.h    filteringedit.h        luaexecutethread.hpp  painterwidget.cpp      qtonly                  selectoutput.cpp      update-myself.lua       wrenchmainwindow.h
adbphonescreenthread.cpp  build-wine.sh       dialoggetentry.ui   filteringlistview.cpp  luamd5                painterwidget.h        qt-solutions            selectoutput.h        upload-all.sh           wrenchmainwindow.ui
adbphonescreenthread.hpp  build-win-mingw.sh  download            filteringlistview.h    macx                  phonescreendialog.cpp  Readme.org              Setclip.apk           vcard.h                 wrench-mime.xml
adbstatethread.cpp        build-win-vc.sh     emacs-weixin.sh     filteringmodel.cpp     main.cpp              phonescreendialog.h    register-wrench.sh      Setclip.apk.md5       window_post_botton.lua  Wrench.pro
adbstatethread.hpp        CallMyLove.twa      emojimodel.cpp      filteringmodel.h       package-all.sh        phonescreendialog.ui   release                 strlistmodel.cpp      windows                 Wrench.qrc
base64.lua                commit-all.sh       emojimodel.h        fix-url.sh             painteritem.cpp       phonescreensyncer.cpp  release-exclude.txt     strlistmodel.h        wrench-get-pictures     Wrench.rc
bhj-alipay.png            config.png          emojis              hide                   painteritem.h         phonescreensyncer.h    Release.org             test.vcf              wrench.h
bhj_help.cpp              contactmodel.cpp    emojis.lua          images                 painterpathitem.cpp   point.png              reset-all.sh            the-true-adb          Wrench.icns
[parker@x3720 Wrench]$ ls -lah adb
lrwxrwxrwx 1 parker parker 27 Jul 11 07:18 adb -> linux/binaries/the-true-adb
[parker@x3720 Wrench]$ ls linux/binaries/the-true-adb 
linux/binaries/the-true-adb
[parker@x3720 Wrench]$ ls -lah linux/binaries/the-true-adb 
lrwxrwxrwx 1 parker parker 51 Jul 11 07:18 linux/binaries/the-true-adb -> ../../../../../system-config/bin/Linux/the-true-adb
````

**Comment by baohaojun** [comment]
Ah, sorry. I must provide the adb for Mac/Windows, because I can't assume that users all have adb pre-installed in their PATH.

Fixed.

You can feel free to replace the the-true-adb with your version, just make sure it's version is higher then 1.0.32 (in fact, adb won't work if more than 1 version of adb is running at the same time).

**Comment by parkerlreed** [comment]
Cheers! 
![1468239737](https://cloud.githubusercontent.com/assets/841440/16730383/a441ef74-4740-11e6-97ce-2169e213dc2b.png)

Language is off and when I click the home button the coordinates seem to be off. 

````
"cmd is input tap 541 1714\r\n"
````

When I tap in the center of the home button. I can make a new issue if needed.

**Comment by parkerlreed** [comment]
Dragging is accurate though (input touchscreen swipe)

**Comment by baohaojun** [comment]
Fixed in 47e7f3f493dcdaafb85f4a109c4c174279de1a6a and 892e5b403c333f7bcb2b220fc82c9b21bc9850b9 I think. It's because the phone I develop on has no virtual key (back/home/menu). With these keys, the phone screen size is bigger than “app size”, and to make my automation script to work, when I send a click command, I need to scale the x/y according to the “app size”. But in the case of phone screen display and click, it's different from automation script click, the sizes must not be scaled according to the app-size.

BTW, Congrats!

**Comment by baohaojun** [comment]
BTW, phone screen display is too slow and will make your phone hot. So please don't abuse it😅.
